src/Base/Service/CurrencyService.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\Base\Service;
+
+class CurrencyService
+{
+    /**
+     * Normalizes an array of any combination of price, vat or priceAfterVat, provided that at least two of the three
+     * are present.
+     *
+     * @param  array{price?: int, vat?: int, priceAfterVat?: int} $prices
+     *
+     * @return array
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     */
+    public function calculateVatTotals(array $prices): array
+    {
+        $price         = $prices['price'] ?? 0;
+        $vat           = $prices['vat'] ?? 0;
+        $priceAfterVat = $prices['priceAfterVat'] ?? 0;
+
+        if ($price && $vat) {
+            $priceAfterVat = $price + $vat;
+        } elseif ($price && $priceAfterVat) {
+            $vat = $priceAfterVat - $price;
+        } elseif ($priceAfterVat && $vat) {
+            $price = $priceAfterVat - $vat;
+        } elseif ($price && ! $priceAfterVat && ! $vat) {
```

Comment:
Deze regel komt ie alleen als die laatste twee 0 zijn, dus die '!' condities hoeven niet, maar dat is misschien voor de duidelijkheid? Ik zou het weglaten.

Reply:
cool, zo werkt ie nog steeds idd :) had het grotendeels aan copilot overgelaten